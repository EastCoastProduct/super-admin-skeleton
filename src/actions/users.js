import queryString from 'query-string';
import { USERS_GET_SUCCESS, USERS_GET_FAILED, USERS_PAGINATION_CHANGE }
  from '../constants/actions';
import { API_URL } from '../constants/application';
import fetch from '../utils/fetch';

export const usersGetSuccess = (list, listTotal) => ({
  type: USERS_GET_SUCCESS,
  list,
  listTotal,
});

export const usersGetFailed = error => ({
  type: USERS_GET_FAILED,
  error,
});

export const paginationChange = page => ({
  type: USERS_PAGINATION_CHANGE,
  page,
});

export const usersGetFetch = (qs, cb) =>
  dispatch =>
    fetch(`${API_URL}/superAdmin/users?${queryString.stringify(qs)}`)
      .then((resp) => {
        dispatch(usersGetSuccess(resp.rows, resp.count));
        return typeof cb === 'function' && cb();
      }).catch(err =>
        dispatch(usersGetFailed(err.message))
      );
