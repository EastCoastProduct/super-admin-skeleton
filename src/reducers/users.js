import { fromJS } from 'immutable';
import { USERS_GET_SUCCESS, USERS_GET_FAILED } from '../constants/actions';

const initialState = fromJS({
  error: null,
  list: [],
});

export default (state = initialState, action) => {
  switch (action.type) {
    case USERS_GET_SUCCESS:
      return state.merge(fromJS({
        error: null,
        list: action.list,
      }));
    case USERS_GET_FAILED:
      return state.merge(fromJS({
        error: action.error,
      }));
    default:
      return state;
  }
};
