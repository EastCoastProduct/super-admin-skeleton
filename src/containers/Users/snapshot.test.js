import React from 'react';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';
import { StyleSheetTestUtils } from 'aphrodite/no-important';
import { list, listTotal, page } from '../../fixtures/users';
import { UsersComponent } from './';

jest.mock('redux-form/immutable', () => ({
  Field: 'Field',
  reduxForm: () => component => component,
}));

jest.mock('../../components/UsersFiltersForm', () => 'UsersFiltersForm');

describe('Users component snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('renders empty required data', () => {
    const tree = renderer.create(
      <UsersComponent
        dispatch={() => {}}
        filters={fromJS({ confirmed: false, search: '' })}
        router={{}}
        users={fromJS({
          error: null,
          list: [],
          listTotal: 0,
          page: 1
        })}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders full required data', () => {
    const tree = renderer.create(
      <UsersComponent
        dispatch={() => {}}
        filters={fromJS({ confirmed: false, search: '' })}
        router={{}}
        users={fromJS({
          error: null,
          list,
          listTotal,
          page
        })}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
