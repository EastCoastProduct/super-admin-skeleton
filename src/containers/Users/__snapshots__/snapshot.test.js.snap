exports[`Users component snapshot renders empty required data 1`] = `
<main
  className="page_3nmjz0">
  <h1
    className="header_1n3lbw7">
    Users
  </h1>
  <div
    className="filters_cljdc4">
    <UsersFiltersForm
      className={
        Object {
          "_definition": Object {
            "display": "inline-block",
            "verticalAlign": "middle",
          },
          "_name": "userFilters_1fsenqq",
        }
      }
      handleFiltersSubmit={[Function]} />
    <a
      className="button_rr5u1q"
      onClick={[Function]}
      style={Object {}}>
      Add User
    </a>
  </div>
</main>
`;

exports[`Users component snapshot renders full required data 1`] = `
<main
  className="page_3nmjz0">
  <h1
    className="header_1n3lbw7">
    Users
  </h1>
  <div
    className="filters_cljdc4">
    <UsersFiltersForm
      className={
        Object {
          "_definition": Object {
            "display": "inline-block",
            "verticalAlign": "middle",
          },
          "_name": "userFilters_1fsenqq",
        }
      }
      handleFiltersSubmit={[Function]} />
    <a
      className="button_rr5u1q"
      onClick={[Function]}
      style={Object {}}>
      Add User
    </a>
  </div>
  <table
    className="table_jphaas">
    <thead>
      <tr
        className="headerRow_1okyf6d">
        <th
          className="cell_jlnq8r">
          First Name
        </th>
        <th
          className="cell_jlnq8r">
          Last Name
        </th>
        <th
          className="cell_jlnq8r">
          Email
        </th>
        <th
          className="cell_jlnq8r">
          Bio
        </th>
        <th
          className="cell_jlnq8r">
          Confirmed
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="row_peo41w"
        onClick={[Function]}>
        <td
          className="cell_jlnq8r">
          regular
        </td>
        <td
          className="cell_jlnq8r">
          regular
        </td>
        <td
          className="cell_jlnq8r">
          regular@mail.com
        </td>
        <td
          className="cell_jlnq8r">
          N/A
        </td>
        <td
          className="cell_jlnq8r">
          true
        </td>
      </tr>
      <tr
        className="row_peo41w"
        onClick={[Function]}>
        <td
          className="cell_jlnq8r">
          name
        </td>
        <td
          className="cell_jlnq8r">
          lastname
        </td>
        <td
          className="cell_jlnq8r">
          user@mail.com
        </td>
        <td
          className="cell_jlnq8r">
          N/A
        </td>
        <td
          className="cell_jlnq8r">
          true
        </td>
      </tr>
      <tr
        className="row_peo41w"
        onClick={[Function]}>
        <td
          className="cell_jlnq8r">
          John
        </td>
        <td
          className="cell_jlnq8r">
          Doe
        </td>
        <td
          className="cell_jlnq8r">
          john@doe.com
        </td>
        <td
          className="cell_jlnq8r">
          N/A
        </td>
        <td
          className="cell_jlnq8r">
          true
        </td>
      </tr>
      <tr
        className="row_peo41w"
        onClick={[Function]}>
        <td
          className="cell_jlnq8r">
          firstname3
        </td>
        <td
          className="cell_jlnq8r">
          lastname3
        </td>
        <td
          className="cell_jlnq8r">
          not.confirmed@mail.com
        </td>
        <td
          className="cell_jlnq8r">
          N/A
        </td>
        <td
          className="cell_jlnq8r">
          false
        </td>
      </tr>
      <tr
        className="row_peo41w"
        onClick={[Function]}>
        <td
          className="cell_jlnq8r">
          firstname4
        </td>
        <td
          className="cell_jlnq8r">
          lastname4
        </td>
        <td
          className="cell_jlnq8r">
          firstname4.lastname4@mail.com
        </td>
        <td
          className="cell_jlnq8r">
          N/A
        </td>
        <td
          className="cell_jlnq8r">
          true
        </td>
      </tr>
      <tr
        className="row_peo41w"
        onClick={[Function]}>
        <td
          className="cell_jlnq8r">
          firstname5
        </td>
        <td
          className="cell_jlnq8r">
          lastname5
        </td>
        <td
          className="cell_jlnq8r">
          forgot.password@mail.com
        </td>
        <td
          className="cell_jlnq8r">
          N/A
        </td>
        <td
          className="cell_jlnq8r">
          true
        </td>
      </tr>
      <tr
        className="row_peo41w"
        onClick={[Function]}>
        <td
          className="cell_jlnq8r">
          firstname6
        </td>
        <td
          className="cell_jlnq8r">
          lastname6
        </td>
        <td
          className="cell_jlnq8r">
          delete.one@mail.com
        </td>
        <td
          className="cell_jlnq8r">
          N/A
        </td>
        <td
          className="cell_jlnq8r">
          true
        </td>
      </tr>
      <tr
        className="row_peo41w"
        onClick={[Function]}>
        <td
          className="cell_jlnq8r">
          firstname7
        </td>
        <td
          className="cell_jlnq8r">
          lastname7
        </td>
        <td
          className="cell_jlnq8r">
          delete.two@mail.com
        </td>
        <td
          className="cell_jlnq8r">
          N/A
        </td>
        <td
          className="cell_jlnq8r">
          true
        </td>
      </tr>
      <tr
        className="row_peo41w"
        onClick={[Function]}>
        <td
          className="cell_jlnq8r">
          firstname8
        </td>
        <td
          className="cell_jlnq8r">
          lastname8
        </td>
        <td
          className="cell_jlnq8r">
          update.one@mail.com
        </td>
        <td
          className="cell_jlnq8r">
          N/A
        </td>
        <td
          className="cell_jlnq8r">
          true
        </td>
      </tr>
      <tr
        className="row_peo41w"
        onClick={[Function]}>
        <td
          className="cell_jlnq8r">
          firstname9
        </td>
        <td
          className="cell_jlnq8r">
          lastname9
        </td>
        <td
          className="cell_jlnq8r">
          confirmed.one@mail.com
        </td>
        <td
          className="cell_jlnq8r">
          N/A
        </td>
        <td
          className="cell_jlnq8r">
          false
        </td>
      </tr>
    </tbody>
    <tfoot
      className="foot_1ntyrs8">
      <tr>
        <td
          colSpan="4">
          <div
            className="pagination_gdz6aw">
            <p>
              Showing all of 10 results
            </p>
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</main>
`;
